### DISCUSSION TITLE
How to extract double from parameter?

### DISCUSSION THREAD

**Discussion by mergesort** [pointfreeco/swift-url-routing#99] [Q&A] [answered]
I have a set of routes that have approximately this shape to them
```swift
extension AppRoute {
    enum ActionsRoute {
        case copy
        case sync(timestamp: Int)
    }
}
```

What I really want though is `case sync(timestamp: Double)`, but I can't for the life of me figure out how to get it to compile. This is the solution I've come up with, but the downside is that a valid timestamp like 1000.01 will become 1000. It's not a huge problem, but it could manifest as issues in other scenarios.

```swift
static let router = OneOf {
    Route(.case(AppRoute.ActionsRoute.copy)) {
        Path { "copy" }

        Query {
            Field("item") { "id" }
        }
    }

    Route(.case(AppRoute.ActionsRoute.sync)) {
        Path { "sync" }

        Query {
            Field("from", default: Int(Date.now.timeIntervalSince1970)) { Digits() }
        }
    }
}

@MainActor
func handleDeepLink(_ route: AppRoute.ActionsRoute) {
    switch route {

    case .copy:
        // Implementation here

    case .sync(let timestamp):
        let timeInterval = Double(timestamp)
        let date = Date(timeIntervalSince1970: timeInterval)

        // Sync from the date above
    }
}
```

I've gone through the code and documentation but couldn't figure out how to properly format the query parameter to use a `Double` in a way that satisfied the compiler, any chance I could get some help figuring that out?

Thank you so much!

**Comment by stephencelis** [comment, marked as answer]
Have you tried `Double.parser()` instead of `Digits()`?

**Reply by mergesort** [reply]
I swore I had tried that but apparently not. 🤦🏻‍♂️ This is of course the correct answer, thank you Stephen!
